sdks/python/apache_beam/transforms/window.py

```diff
@@ -341,14 +343,18 @@ def __ne__(self, other):
 
   def to_runner_api_parameter(self, context):
     return (urns.FIXED_WINDOWS_FN,
-            proto_utils.pack_Struct(size=self.size.micros,
-                                    offset=self.offset.micros))
-
-  @urns.RunnerApiFn.register_urn(urns.FIXED_WINDOWS_FN, struct_pb2.Struct)
+            standard_window_fns_pb2.FixedWindowsPayload(
+                size=proto_utils.from_micros(
+                    duration_pb2.Duration, self.size.micros),
+                offset=proto_utils.from_micros(
+                    timestamp_pb2.Timestamp, self.offset.micros)))
+
+  @urns.RunnerApiFn.register_urn(
```

Comment:
One design decision on the Java side (which I don't feel wedded to) is that URNs and payloads are entirely divorced from their Java native classes. I had imagined adding `PTransform.getUrn()` but instead we have a registrar from Java class to URN and payload which lives entirely outside the core SDK to keep proto off the API surface.

I don't entirely understand the ramifications of the Python SDK making both of these decisions separately. (PTransform having their URN, proto messages being right there too). I'd love your thoughts on it sometime, and whether we can easily make Java more natural.